packages/react/src/components/TreeView/TreeNode.tsx

```diff
@@ -5,23 +5,95 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, { useState, useEffect, useRef } from 'react';
-import PropTypes from 'prop-types';
-import { CaretDown } from '@carbon/icons-react';
+import { CarbonIconType, CaretDown } from '@carbon/icons-react';
 import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useEffect, useRef, useState } from 'react';
 import { keys, match, matches } from '../../internal/keyboard';
-import uniqueId from '../../tools/uniqueId';
-import { usePrefix } from '../../internal/usePrefix';
 import { useControllableState } from '../../internal/useControllableState';
+import { usePrefix } from '../../internal/usePrefix';
+import uniqueId from '../../tools/uniqueId';
 import { useFeatureFlag } from '../FeatureFlags';
 
-const TreeNode = React.forwardRef(
+export type TreeNodeProps = {
+  /**
+   * **Note:** this is controlled by the parent TreeView component, do not set manually.
+   * The ID of the active node in the tree
+   */
+  active?: string | number;
+  /**
+   * Specify the children of the TreeNode
+   */
+  children?: React.ReactNode;
+  /**
+   * Specify an optional className to be applied to the TreeNode
+   */
+  className?: string;
+  /**
+   * **[Experimental]** The default expansion state of the node.
+   * *This is only supported with the `enable-treeview-controllable` feature flag!*
+   */
+  defaultIsExpanded?: boolean;
+  /**
+   * **Note:** this is controlled by the parent TreeView component, do not set manually.
+   * TreeNode depth to determine spacing
+   */
+  depth?: number;
+  /**
+   * Specify if the TreeNode is disabled
+   */
+  disabled?: boolean;
+  /**
+   * Specify the TreeNode's ID. Must be unique in the DOM and is used for props.active and props.selected
+   */
+  id?: string;
+  /**
+   * Specify if the TreeNode is expanded (only applicable to parent nodes)
+   */
+  isExpanded?: boolean;
+  /**
+   * Rendered label for the TreeNode
+   */
+  label: React.ReactNode;
+  /**
+   * Callback function for when the node receives or loses focus
+   */
+  onNodeFocusEvent?: (event: React.FocusEvent<HTMLLIElement>) => void;
+  /**
+   * Callback function for when the node is selected
+   */
+  onSelect?: (event: React.MouseEvent, node?: TreeNodeProps) => void;
+  /**
+   * Callback function for when a parent node is expanded or collapsed
+   */
+  onToggle?: (event: React.MouseEvent, node?: TreeNodeProps) => void;
+  /**
+   * Callback function for when any node in the tree is selected
+   */
+  onTreeSelect?: (event: React.MouseEvent, node?: TreeNodeProps) => void;
+  /**
+   * Optional prop to allow each node to have an associated icon.
+   * Can be a React component class
+   */
+  renderIcon?: CarbonIconType;
```

Comment:
This is too narrow, it should accept any class or functional component. Throughout the codebase we've been doing `ComponentType | FunctionComponent` or `React.ElementType`. Either should resolve the ts errors on the proptypes I believe.

Reply:
Fixed!